test/forms/inputs/textarea.test.js

```diff
@@ -72,9 +72,24 @@ test('Textrea does not receive invalid dom attributes', () => {
     meta: {},
     maxLength: 5,
     hideCharacterCount: true,
-    onClickLabel: () => 'foo'
+    onClickLabel: () => 'foo',
   }
-  
+
   const wrapper = mount(<Textarea {...props} />)
   expect(wrapper.find('textarea').prop('onClickLabel')).toBe(undefined)
 })
+
+test('Textarea passes down forwardedRef to input correctly', () => {
```

Comment:
Similarly to the story comment, I'd like to see a test for `ref` getting passed directly to TextArea

Reply:
Good thing you called this out! A standard `ref` was **not** working correctly. The issue is we're wrapping `Textarea` in another component when composing with `blurDirty`. I ended up exporting that using `forwardRef` as well to pass the `ref` down to `Textarea`. Do you think that requires some additional new documentation on `blurDirty` to call that out? Any other ideas that might be cleaner?

Reply:
Oof 😬 . The challenge with HOCs is that it looks like for the `ref=` scenario to work, every single HOC would need to have `forwardRef` applied to it. It looks like the [React team's recommendation](https://reactjs.org/docs/forwarding-refs.html#forwarding-refs-in-higher-order-components) is to just use a custom prop name (the same name we choose actually) to pass down to the wrapped component.

Since we do have components that use multiple HOCs (e.g., `checkbox` uses `blurDirty` **and** `replaceEmptyStringValue`), I think it would actually be better to only accept the custom `forwardedRef` prop in `Textarea` (i.e., remove the `forwardRef()` usages, stories, and tests).

That being said, in the long run we could create a `withForwardedRef` HOC to apply to other inputs that would pass the custom ref prop down automatically (example below). There might also be other alternatives that we could explore by taking a look at some more established OS react component libraries. I would definitely consider all of that out of scope for this PR. As an aside, it always looks like the React team recommends the inclusion of `forwardRef` to warrant a major version change as it should be considered a breaking change (https://reactjs.org/docs/forwarding-refs.html#note-for-component-library-maintainers).

```js
function withForwardedRef(Component) {
  return React.forwardRef((props, ref) => <Component {...props} forwardedRef={ref} />)
}
```
Sorry for the reversal on this decision, but hopefully you found this exploration informative!

Reply:
All good, it was helpful to walk through how it would with HOCs. Ready for another look!